community/kernel/src/main/java/org/neo4j/kernel/impl/api/store/StorePropertyCursor.java

```diff
@@ -78,14 +80,15 @@
     private int offsetAtBeginning;
     private int blocks;
     private int remainingBlocksToRead;
-    private long header;
     private PropertyType type;
     private int keyId;
     private ByteBuffer buffer;
 
     private AbstractDynamicStore.DynamicRecordCursor recordCursor;
     private long currentRecordId;
-    private long[] valuesAfterHeader;
+
+    // in values[0] will be placed the property block header
+    private long[] values = new long[VALUES_SIZE];
```

Comment:
that will a bit improve value[0], but value[1] still be there :)

what about wrapping `values` into helper class and provide `header()`, `toBits()`, `remainingHead()` and `setRemaining(index)` or similar there. That will keep a way how data can be copied into buffer and will hide from `StorePropertyCursor ` all array indexing juggling? WDYT?